### PR TITLE
feat: Adding ci/cd benchmark

### PR DESCRIPTION
**Pull request by byawitz**
## What does this PR do?

Adding CI/CD benchmark on PR

### CODE REVIEW

File: .github/workflows/benchmark.yml [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+      - name: Prepare comment
+        run: |
+          echo '## :sparkles: Benchmark results' > benchmark.txt
+          echo ' ' >> benchmark.txt
+          echo "- Requests per second: $(jq -r '.summary.requestsPerSec|tonumber?|floor' benchmark.json)" >> benchmark.txt
+          echo "- Total 200 requests: $(jq -r '.statusCodeDistribution."200"' benchmark.json)" >> benchmark.txt
+          echo "- 99 latency: $(jq -r '.latencyPercentiles.p99' benchmark.json )" >> benchmark.txt
+          echo " " >> benchmark.txt
+          echo " " >> benchmark.txt
+          echo "## :zap:  Benchmark Compering" >> benchmark.txt
```
Comment (christyjacob4) [inline .github/workflows/benchmark.yml:52]:
```suggestion
          echo "## :zap:  Benchmark Comparison" >> benchmark.txt
```

File: .github/workflows/benchmark.yml [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+          echo " " >> benchmark.txt
+          echo "| Metric | PR | Latest version | "  >> benchmark.txt
```
Comment (christyjacob4) [inline .github/workflows/benchmark.yml:54]:
```suggestion
          echo "| Metric | This PR | Latest version | "  >> benchmark.txt
```

File: .github/workflows/benchmark.yml [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+          sleep 10
+      - name: Benchmark Latest
+        run: oha -z 180s  http://localhost/v1/health/version -j > benchmark-latest.json
+      - name: Prepare comment
+        run: |
+          echo '## :sparkles: Benchmark results' > benchmark.txt
+          echo ' ' >> benchmark.txt
+          echo "- Requests per second: $(jq -r '.summary.requestsPerSec|tonumber?|floor' benchmark.json)" >> benchmark.txt
+          echo "- Total 200 requests: $(jq -r '.statusCodeDistribution."200"' benchmark.json)" >> benchmark.txt
+          echo "- 99 latency: $(jq -r '.latencyPercentiles.p99' benchmark.json )" >> benchmark.txt
```
Comment (christyjacob4) [inline .github/workflows/benchmark.yml:49]:
```suggestion
          echo "- P99 latency: $(jq -r '.latencyPercentiles.p99' benchmark.json )" >> benchmark.txt
```

File: .github/workflows/benchmark.yml [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+          docker compose up -d
+          sleep 10
+      - name: Benchmark Latest
+        run: oha -z 180s  http://localhost/v1/health/version -j > benchmark-latest.json
+      - name: Prepare comment
+        run: |
+          echo '## :sparkles: Benchmark results' > benchmark.txt
+          echo ' ' >> benchmark.txt
+          echo "- Requests per second: $(jq -r '.summary.requestsPerSec|tonumber?|floor' benchmark.json)" >> benchmark.txt
+          echo "- Total 200 requests: $(jq -r '.statusCodeDistribution."200"' benchmark.json)" >> benchmark.txt
```
Comment (christyjacob4) [inline .github/workflows/benchmark.yml:48]:
```suggestion
          echo "- Requests with 200 status code: $(jq -r '.statusCodeDistribution."200"' benchmark.json)" >> benchmark.txt
```

**Review by christyjacob4 — APPROVED** [review]

File: .github/workflows/benchmark.yml [hunk 5]
```diff
@@ -0,0 +1,69 @@
+name: "Benchmarks"
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
```
Comment (christyjacob4) [inline .github/workflows/benchmark.yml:6]: Can we make this a little more optimal and make the benchmark part of the existing tests here https://github.com/appwrite/appwrite/blob/main/.github/workflows/tests.yml#L41-L47

That way we already have the image built and saved in cache. 

We can add a new job called `benchmark` that uses the cached image from the `setup` job.

Comment (byawitz) [reply .github/workflows/benchmark.yml]: Done

**Review by christyjacob4 — APPROVED** [review]

**Pull request merged**